### ISSUE TITLE
How configuration path roles CustomCasSecurityConfiguration HttpSecurity

### ISSUE THREAD

**Issue by liyiorg** [kakawait/cas-security-spring-boot-starter#67]
hello,I want configuration path authorize must has role! I override  method configure(HttpSecurity http), 
User login and have roles  'A' ,  but  the user can access any paths!

thanks!

```java
@Configuration
class CustomCasSecurityConfiguration extends CasSecurityConfigurerAdapter {
	
	@Autowired
	private List<CasSecurityConfigurer> configurers;
	
	@Override
	public void configure(CasAuthenticationFilterConfigurer filter) {
		// Here you can configure CasAuthenticationFilter
	}

	@Override
	public void configure(CasSingleSignOutFilterConfigurer filter) {
		// Here you can configure SingleSignOutFilter
	}

	@Override
	public void configure(CasAuthenticationProviderSecurityBuilder provider) {
		// Here you can configure CasAuthenticationProvider
	}

	@Override
	public void configure(HttpSecurity http) throws Exception {
		// Here you can configure Spring Security HttpSecurity object during
		// init configure
		http
			.authorizeRequests()
				.antMatchers("/testa").hasRole("A")
				.antMatchers("/testb").hasRole("B");
	}

	@Override
	public void configure(CasTicketValidatorBuilder ticketValidator) {
		// Here you can configure CasTicketValidator
	}
}

```

spring boot start log
```java
2017-12-27 15:43:27.091 DEBUG 11056 --- [  restartedMain] edFilterInvocationSecurityMetadataSource : Adding web access control expression 'hasRole('ROLE_USER')', for org.springframework.security.web.util.matcher.AnyRequestMatcher@1
2017-12-27 15:43:27.091 DEBUG 11056 --- [  restartedMain] edFilterInvocationSecurityMetadataSource : Adding web access control expression 'hasRole('ROLE_A')', for Ant [pattern='/testa']
2017-12-27 15:43:27.092 DEBUG 11056 --- [  restartedMain] edFilterInvocationSecurityMetadataSource : Adding web access control expression 'hasRole('ROLE_B')', for Ant [pattern='/testb']
```

**Comment by kakawait** [comment]
@liyiorg thank you for reporting i will check that asap but I'm currently on vacation. So I'm not sure to do something before January. Sorry

**Comment by kakawait** [comment]
@liyiorg I fixed something on _develop_ branch. I will push asap `0.6.1-SNAPSHOT` on maven central (I will say you when it will be available) in order to test the fix if you're ok.

**Comment by liyiorg** [comment]
@kakawait  I change project dependency to 0.6.1-SNAPSHOT,Test in two ways.
user list
* usera  has roles A
* userb  has roles B

Cas client service config
```yam
server:
  port: 9090
security:
  cas:
    server:
      base-url: https://localhost:8443/cas
    service:
      resolution-mode: dynamic
    user:
      roles-attributes:
      - roles
```


## test 1 
use @Secured annotation,it's work ok.   
The user "usera" only access path /testa .
The user "userb" only access path /testb.

```java
@Controller
public class TestController {

	@Secured("ROLE_A")
	@RequestMapping("/testa")
	public String testa(HttpServletRequest request){
		System.out.println("hasRoleA:" + request.isUserInRole("A"));
		return "test_cas";
	}
	
	@Secured("ROLE_B")
	@RequestMapping("/testb")
	public String testb(HttpServletRequest request){
		System.out.println("hasRoleB:" + request.isUserInRole("B"));
		return "test_cas";
	}
}
```

## test 2 
Override method configure(HttpSecurity http) and remove @Secured annotation.

The usera access path http://localhost:9090/testa
```log
print  hasRoleA:true
```
The usera access path http://localhost:9090/testb
```log
print  hasRoleB:false
```
The user usera can access any path!

```java
@Configuration
class CustomCasSecurityConfiguration extends CasSecurityConfigurerAdapter {
	
	@Override
	public void configure(CasAuthenticationFilterConfigurer filter) {
		// Here you can configure CasAuthenticationFilter
	}

	@Override
	public void configure(CasSingleSignOutFilterConfigurer filter) {
		// Here you can configure SingleSignOutFilter
	}

	@Override
	public void configure(CasAuthenticationProviderSecurityBuilder provider) {
		// Here you can configure CasAuthenticationProvider
	}

	@Override
	public void configure(HttpSecurity http) throws Exception {
		// Here you can configure Spring Security HttpSecurity object during
		// init configure
		http
			.authorizeRequests()
				.antMatchers("/testa").hasRole("A")
				.antMatchers("/testb").hasRole("B");
	}

	@Override
	public void configure(CasTicketValidatorBuilder ticketValidator) {
		// Here you can configure CasTicketValidator
	}
}

```
```java
@Controller
public class TestController {

	//@Secured("ROLE_A")
	@RequestMapping("/testa")
	public String testa(HttpServletRequest request){
		System.out.println("hasRoleA:" + request.isUserInRole("A"));
		return "test_cas";
	}
	
	//@Secured("ROLE_B")
	@RequestMapping("/testb")
	public String testb(HttpServletRequest request){
		System.out.println("hasRoleB:" + request.isUserInRole("B"));
		return "test_cas";
	}
}
```

**Comment by kakawait** [comment]
@liyiorg you said

> I change project dependency to 0.6.1-SNAPSHOT

And it works without any other operations? Because you have to add snapshot repository or `mvn clean install` project in _develop_ branch.

I asking, I'm not able to reproduce. (Just I don't use `security.cas.user.roles-attribute.roles` I will try with that asap)

**Comment by kakawait** [comment]
@liyiorg could you please tell me how do you upgrade to `0.6.1-SNAPSHOT`? It may missing something

**Comment by liyiorg** [comment]
@kakawait , 0.6.1-SNAPSHOT run ok.
 thank you very much for your support and the unselfish contribution of open source!

**Comment by kakawait** [comment]
@liyiorg Nice I will release version really soon

I'm happy when everyone are happy :)

**Comment by liyiorg** [comment]
@kakawait   When will it be released  0.6.1-release

**Comment by kakawait** [comment]
Sorry I get busy i will release it todat
